### PR TITLE
Changed file share mode for attachment upload

### PR DESCRIPTION
**Pull request by ShreyasRmsft**
PR for issue https://github.com/Microsoft/azure-pipelines-agent/issues/2034

### CODE REVIEW

File: src/Microsoft.VisualStudio.Services.Agent/JobServerQueue.cs [hunk 1]
```diff
@@ -644,7 +644,7 @@ private async Task UploadFile(UploadFileInfo file)
                 else
                 {
                     // Create attachment
-                    using (FileStream fs = File.Open(file.Path, FileMode.Open, FileAccess.Read, FileShare.Read))
+                    using (FileStream fs = File.Open(file.Path, FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
```
Comment (TingluoHuang) [inline src/Microsoft.VisualStudio.Services.Agent/JobServerQueue.cs]: why needs to be ReadWrite?

Comment (TingluoHuang) [reply src/Microsoft.VisualStudio.Services.Agent/JobServerQueue.cs]: change line 635 as well.

Comment (TingluoHuang) [inline src/Microsoft.VisualStudio.Services.Agent/JobServerQueue.cs]: do same thing on line 635

Comment (ShreyasRmsft) [reply src/Microsoft.VisualStudio.Services.Agent/JobServerQueue.cs]: done

**Pull request merged**